### PR TITLE
Add audit log rate limit category and make rate limit category getter public

### PR DESCRIPTION
**Pull request by Letiste**
When calling `https://api.github.com/rate_limit`, the endpoint returns a `audit_log` rate limit category as well. This is used for `/audit-log` endpoints:
- https://docs.github.com/en/enterprise-cloud@latest/rest/orgs/orgs?apiVersion=2022-11-28#get-the-audit-log-for-an-organization
- https://docs.github.com/en/enterprise-cloud@latest/admin/monitoring-activity-in-your-enterprise/reviewing-audit-logs-for-your-enterprise/using-the-audit-log-api-for-your-enterprise#using-the-audit-log-api

This PR adds this new rate limit category.

I have a use case where I need to know the rate limit category before making the request so as to not reimplement what's already done in `go-github`, I made this logic public. Other users could also benefit of this change.
If you think it's not reasonable and it should be kept private, I will revert this change.

### CODE REVIEW

**Comment by Letiste** [comment]
@gmlewis could you take a look at this when you have some time? 🙏 (I'm notifying you as you seem to review the recent PRs, sorry if you are not the correct person for that).
Let me know if I need to clarify some things.
Not sure as well why the codecov is failing here, I thought I added all the necessary tests related to my changes but maybe I missed some

**Review by gmlewis — APPROVED** [review]
Thank you, @Letiste !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by rpelliard — APPROVED** [review]

**Comment by gmlewis** [comment]
Thank you, @rpelliard !
Merging.

**Pull request merged**
